paddle/fluid/framework/op_registry.h

```diff
@@ -90,15 +91,20 @@ class OpRegistry {
    *            attr_check is set to false, otherwise it will be failed
    *            when check function called.
    */
-  static std::unique_ptr<OperatorBase> CreateOp(const std::string& type,
-                                                const VariableNameMap& inputs,
-                                                const VariableNameMap& outputs,
-                                                AttributeMap attrs,
-                                                bool attr_check = true);
-
-  static std::unique_ptr<OperatorBase> CreateOp(const proto::OpDesc& op_desc);
+  static std::unique_ptr<OperatorBase> CreateOp(
+      const std::string& type, const VariableNameMap& inputs,
+      const VariableNameMap& outputs, AttributeMap attrs,
+      const std::vector<std::string>& input_names = {},
```

Comment:
maybe if we write this way, we can solve the current problem, but it may increase the cost of reconstruction in the future

Reply:
This is a way to `add` some features without changing existing operators. I agree that refactoring `VariableNameMap` may be a better way but it requires modifying all operators. This way is a trade-off.